app/design/adminhtml/default/default/template/sales/order/creditmemo/create/items.phtml

```diff
@@ -104,6 +104,10 @@
               <label class="normal" for="notify_customer"><?php echo Mage::helper('sales')->__('Append Comments') ?></label>
               <input id="notify_customer" name="creditmemo[comment_customer_notify]" value="1" type="checkbox" />
           </p>
+          <p>
```

Comment:
Why not first check that customer is subscribed and only show this option if so?